### ISSUE TITLE
test_busted: float() on requires_grad tensors emits UserWarning in test_forward_output_ranges

### ISSUE THREAD

**Issue by nekrut** [veg/HyphAeon#30]
### Summary

`tests/test_busted.py::TestBustedMultiTaskHead::test_forward_output_ranges` calls `float(...)` directly on tensors that still carry `requires_grad=True`, which triggers a `UserWarning` from PyTorch on every run:

```
tests/test_busted.py:38: UserWarning: Converting a tensor with requires_grad=True to a scalar
may lead to unexpected behavior. Consider using tensor.detach() first.
```

### Location

`tests/test_busted.py`, in `test_forward_output_ranges`:

```python
out = head(x)
assert 0.0 <= float(out["cls_prob"]) <= 1.0
assert float(out["pred_lrt"]) >= 0.0
assert float(out["pred_logp"]) >= 0.0
assert float(out["pred_omega3"]) >= 0.0
assert float(out["syn_var"]) >= 0.0
```

The head is run without `torch.no_grad()`, so these outputs require grad, and `float()` on them warns. Note the sibling assertion in the same test already does the right thing for `omega_prop` by calling `.detach()` first.

### Suggested fix

Either wrap the forward in `torch.no_grad()`:

```python
with torch.no_grad():
    out = head(x)
```

or `.detach()` each scalar before converting. Cosmetic only — the test passes; this just removes warning noise from the test output.

### Environment

Observed with `torch 2.14.0`, Python 3.11, running `python -m pytest tests/ -v` (`241 passed, 2 skipped, 2 warnings`).
